referenceframe/frame.go

```
@@ -445,3 +445,117 @@ func (rf rotationalFrame) MarshalJSON() ([]byte, error) {
+	}
+	return spatial.NewPose(
+		r3.Vector{X: inputs[0].Value, Y: inputs[1].Value, Z: inputs[2].Value},
+		&spatial.OrientationVector{
+			OX:    inputs[3].Value,
+			OY:    inputs[4].Value,
+			OZ:    inputs[5].Value,
+			Theta: inputs[6].Value,
+		},
+	), nil
+}
+
+// Interpolate interpolates the given amount between the two sets of inputs.
+func (pf *poseFrame) Interpolate(from, to []Input, by float64) ([]Input, error) {
+	if err := pf.baseFrame.validInputs(from); err != nil {
+		return nil, NewIncorrectInputLengthError(len(from), 7)
+	}
+	if err := pf.baseFrame.validInputs(to); err != nil {
+		return nil, NewIncorrectInputLengthError(len(to), 7)
+	}
+	fromPose, err := pf.Transform(from)
+	if err != nil {
+		return nil, err
+	}
+	toPose, err := pf.Transform(to)
+	if err != nil {
+		return nil, err
+	}
+	interpolatedPose := spatial.Interpolate(fromPose, toPose, by)
+	return []Input{
+		{interpolatedPose.Point().X},
+		{interpolatedPose.Point().Y},
+		{interpolatedPose.Point().Z},
+		{interpolatedPose.Orientation().OrientationVectorRadians().OX},
+		{interpolatedPose.Orientation().OrientationVectorRadians().OY},
+		{interpolatedPose.Orientation().OrientationVectorRadians().OZ},
+		{interpolatedPose.Orientation().OrientationVectorRadians().Theta},
+	}, nil
+}
+
+// Geometries returns an object representing the 3D space associeted with the staticFrame.
+func (pf *poseFrame) Geometries(inputs []Input) (*GeometriesInFrame, error) {
+	transformByPose, err := pf.Transform(inputs)
+	if err != nil {
+		return nil, err
+	}
+	if pf.geometry == nil {
+		return NewGeometriesInFrame(pf.name, []spatial.Geometry{}), nil
+	}
+	return NewGeometriesInFrame(pf.name, []spatial.Geometry{pf.geometry.Transform(transformByPose)}), nil
+}
+
+// DoF returns the number of degrees of freedom within a model.
+func (pf *poseFrame) DoF() []Limit {
+	return pf.limits
+}
+
+// MarshalJSON serializes a Model.
+func (pf *poseFrame) MarshalJSON() ([]byte, error) {
```

I don't believe that this is valid, as it will not deserialize. Instead this should return an error for this method until we fully implement serialization for it.